### ISSUE TITLE
[Phase 2] Translation LoRA training on Vast.ai

### ISSUE THREAD

**Issue by selcuk-yalcin** [selcuk-yalcin/Kurdish-LLM-FineTuning#5]
## Summary

Fine-tune a **LoRA adapter** on the translation dataset (EN/TR → Kurmanji Latin) using a rented **Vast.ai GPU**, then upload the adapter to Hugging Face Hub.

## Pipeline

```
HF Hub (data) → Vast.ai (train) → HF Hub (adapter) → Mac Gradio (test)
```

## Prerequisites

- [ ] [#1](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/issues/1) HF repos verified
- [ ] [#2](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/issues/2) Vast.ai account + bootstrap tested
- [ ] [#3](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/issues/3) Data spot-check passed
- [ ] [#4](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/issues/4) LoRA/QLoRA config finalized

### On Vast.ai
- [ ] Rent GPU: RTX 4090 (24 GB), ≥ 50 GB disk
- [ ] Bootstrap: `bash scripts/vastai/bootstrap.sh`
- [ ] **Smoke test:** 100 training steps — verify loss decreases, no OOM
- [ ] **Full run:** `bash scripts/vastai/train_translation.sh`
- [ ] Evaluate on held-out EN/TR eval sets (`data/translation/eval/`)
- [ ] Upload adapter: auto via script → `selcukyalcin/kurdish-translation-lora`
- [ ] Save training logs to `logs/`
- [ ] **Destroy instance**

### Evaluation (can split with Cem)
- [ ] Run `scripts/translation/run_inference.py` on eval set
- [ ] Measure: Kurmanji output rate (≥95%), Latin script compliance (≥98%)
- [ ] Optional: BLEU/chrF vs reference pairs
- [ ] @cem-acun: subjective quality review on 20 eval samples

## Acceptance criteria

- LoRA adapter on HF Hub (`kurdish-translation-lora`)
- Eval metrics logged
- Gradio loads adapter from Hub successfully

## Parallel work

**Sequential dependency:** Requires Phase 0 + Phase 1 complete.

**During training:** @cem-acun can start **TTS dataset research** ([#6](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/issues/6)) — fully independent track.

## Resources

- [scripts/vastai/train_translation.sh](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/blob/main/scripts/vastai/train_translation.sh)
- [src/translation/trainer.py](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/blob/main/src/translation/trainer.py)
- [config/translation/](https://github.com/selcuk-yalcin/Kurdish-LLM-FineTuning/tree/main/config/translation)
- [PEFT LoRA docs](https://huggingface.co/docs/peft/main/en/conceptual_guides/lora)

## Assignee

@selcuk-yalcin — training run on Vast.ai  
@cem-acun — eval quality review (parallel, after adapter uploaded)

**Comment by selcuk-yalcin** [comment]
Consolidated into streamlined 35-issue roadmap (Aug 2026).
